common/fee/models.go

```diff
@@ -0,0 +1,100 @@
+package fee
+
+import (
+	"math/big"
+
+	"github.com/pkg/errors"
+
+	"github.com/smartcontractkit/chainlink/v2/common/chains/label"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	bigmath "github.com/smartcontractkit/chainlink/v2/core/utils/big_math"
+)
+
+var (
+	ErrBumpFeeExceedsLimit = errors.New("fee bump exceeds limit")
+	ErrBump                = errors.New("fee bump failed")
+	ErrConnectivity        = errors.New("transaction propagation issue: transactions are not being mined")
+)
+
+func IsBumpErr(err error) bool {
+	return err != nil && (errors.Is(err, ErrBumpFeeExceedsLimit) || errors.Is(err, ErrBump) || errors.Is(err, ErrConnectivity))
+}
+
+// CalculateFee computes the fee price and chain specific fee limit for a transaction.
+func CalculateFee(
+	maxFeePrice, defaultPrice, maxBumpPrice *big.Int,
+) (feePrice *big.Int) {
+	feePrice = CapFeePrice(defaultPrice, maxFeePrice, maxBumpPrice)
+	return
```

Comment:
Any more simplification available here?

Reply:
thanks for the catches, will look for other simplifications as well